src/tests/intg/files_ops.py

```diff
@@ -77,6 +77,7 @@ def _read_contents(self):
 
     def _write_contents(self, contents):
         tmp_file = tempfile.NamedTemporaryFile(dir=self.tmp_dir, delete=False)
```

Comment:
If content is unicode, then instead of encoding, NamedTemporaryFile can be opened in textual write mode "w", default mode is binary "w+b"

Would be nice to use `with` keyword with files, but unrelated to this patch 
```python
with tempfile.NamedTemporaryFile("w", dir=self.tmp_dir, delete=False) as tmp_file:
     tmp_file.writelines(contents)
     tmp_file.flush()
```